### PR TITLE
feature: Support profiler config in the pipeline training job step

### PR DESCRIPTION
**Pull request by navaj0**
issue: #2150
test: tox tests/unit

*Issue #, if available:* #2150

*Description of changes:* add support of profiler config in pipeline training step

*Testing done:* `tox tests/unit`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/steps.py [hunk 1]
```diff
@@ -160,9 +160,6 @@ def arguments(self) -> RequestType:
         NOTE: The CreateTrainingJob request is not quite the args list that workflow needs.
         The TrainingJobName and ExperimentConfig attributes cannot be included.
         """
-        self.estimator.disable_profiler = True
-        self.estimator.profiler_config = None
-        self.estimator.profiler_rules = None
```
Comment (ChoiByungWook) [inline src/sagemaker/workflow/steps.py:165]: Are there any edge cases that can arise by having it enabled by default?

Comment (navaj0) [reply src/sagemaker/workflow/steps.py:165]: I don't think so. We enable the profiler by default if a user launch a training job directly by doing `estimator.fit()`. We could do the same for pipeline users.

Comment (ChoiByungWook) [reply src/sagemaker/workflow/steps.py:165]: Can we add an integ test?

Comment (navaj0) [reply src/sagemaker/workflow/steps.py:165]: Yes. Updated the existing test to verify the profiler config.

Comment (ChoiByungWook) [reply src/sagemaker/workflow/steps.py:165]: Awesome! Does this work in every region?

Comment (navaj0) [reply src/sagemaker/workflow/steps.py:165]: Yes. Profiler config is not available in CN, or GovCloud, pipeline is not available in those regions either.

**Review by ChoiByungWook — APPROVED** [review]

**Pull request merged**